core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1096,6 +1102,9 @@ class Partition(val topicPartition: TopicPartition,
    * advancing the HW, the follower's log end offset may keep falling behind the HW (determined by the leader's log end
    * offset) and therefore will never be added to ISR.
    *
+   * Note, If KIP-966 is enabled, the HW can only advance if the ISR size is equal or large than the
+   * min ISR(min.insync.replicas).
```

Comment:
how about "larger than the min ISR"

Reply:
When ISR size is equal to min ISR, the HW can advance.